DFC.App.DiscoverSkillsCareers.Services/SessionIdToCodeConverters/SessionIdToCodeConverter.cs

```diff
@@ -0,0 +1,48 @@
+﻿using DFC.App.DiscoverSkillsCareers.Services.Contracts;
+using System.Text;
+
+namespace DFC.App.DiscoverSkillsCareers.Services.SessionIdToCodeConverters
+{
+    public class SessionIdToCodeConverter : ISessionIdToCodeConverter
+    {
+        public string GetCode(string value)
+        {
+            var result = new StringBuilder();
+            if (!string.IsNullOrWhiteSpace(value))
+            {
+                value = value.Trim().ToUpper();
+                int i = 0;
+                foreach (var c in value)
+                {
+                    i++;
+                    if (i % 4 == 1 && i > 1)
+                    {
+                        result.Append(" ");
+                    }
+
+                    result.Append(c);
+                }
+            }
+
+            return result.ToString();
+        }
+
+        public string GetSessionId(string code)
+        {
+            var result = new StringBuilder();
+
+            if (!string.IsNullOrWhiteSpace(code))
```

Comment:
Not sure about this.  Wouldn't have been better to split the string by a space and then read the given array.  Would be a lot faster.

Reply:
Modified code